### PR TITLE
add option to specify ipv4 addresses

### PR DESCRIPTION
**Pull request by carterjones**
This adds an option to specify a list of IPv4 addresses to be associated with an NLB instead of using a list of elastic IP addresses. This is useful in cases where an internal NLB is required.

This is a breaking change since the the output variable `nlb_elastic_ips` was renamed to `nlb_ips`. I noted this in the readme.

I also made CircleCI tests start to run again, upgraded the CircleCI docker image, bumped pre-commit hook versions (and disabled the `no-inline-html` markdownlint setting), and made pre-commit show diffs if it ever produced them.

Lastly, in the original test, I added a `depends_on` link to ensure that the NLB was created after the EIP and to ensure the reverse was true (that the EIPs would not be deleted until the NLB was created); without this, I had received authorization errors indicating a failure to delete one or more EIPs when CircleCI ran.

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -8,13 +8,13 @@ output "nlb_dns_name" {
...
-output "nlb_elastic_ips" {
-  description = "List of public Elastic IP addresses associated with the NLB."
+output "nlb_ips" {
+  description = "List of IP addresses associated with the NLB."
 
   value = [
-    data.aws_eip.nlb_eip1.public_ip,
-    data.aws_eip.nlb_eip2.public_ip,
-    data.aws_eip.nlb_eip3.public_ip,
+    local.use_eips ? data.aws_eip.nlb_eip1[0].public_ip : local.use_ipv4_addrs ? var.nlb_ipv4_addrs[0] : "undefined",
```
Comment (carterjones) [inline outputs.tf:15]: I'm not sure if this is the right approach or if I should do something like this to simplify it. I'm interested in folks opinions on this.
```terraform
local.use_eips ? data.aws_eip.nlb_eip1[0].public_ip : var.nlb_ipv4_addrs[0]
```

Comment (dynamike) [reply outputs.tf:15]: This seems okay, but I would expect the module to be pretty broken in other ways if both `use_eips` and `use_ipv4_addrs` are false.

**Review by dynamike — APPROVED** [review]
looks good! thanks for all the updates

File: outputs.tf (hunk 1 above)
Comment (carterjones) [reply outputs.tf:15]: @dynamike Agreed. I tried to capture that idea in the readme ("If neither option is defined, the module will fail."). Do you think it warrants further error checking or should we just let it fail on its own? Not sure if there is a best practice with that.

Comment (dynamike) [reply outputs.tf:15]: Error checking in Terraform is fairly limited, you can do validation checking of variables, but not much else. I think what you have now is fine. 

https://www.terraform.io/docs/language/functions/can.html

Comment (carterjones) [reply outputs.tf:15]: Thanks for the link. I didn't know about `can` or `try`.

**Pull request merged**